Source/Base/ASBaseDefines.h

```diff
@@ -232,6 +232,26 @@
   ((c *) ([__val class] == [c class] ? __val : nil));\
 })
 
+// Compare two primitives, assign if different. Returns whether the assignment happened.
+#define ASCompareAssign(lvalue, newValue) ({  \
```

Comment:
Shouldn't there be a variant that does use ASObjectIsEqual, but doesn't do a copy (just a regular assignment which may involve a retain?)

Is there any way to enforce that ASCompareAssign is *not* used for objects accidentally?

Reply:
Good point, I added that variant as `ASCompareAssignObject`.

I think there's value in using plain ASCompareAssign for objects, in the case where we want to skip the `isEqual:` (say, in `setDelegate:` or similar), so I'd rather keep that option open.